core/src/main/java/org/ehcache/core/config/ResourcePoolsImpl.java

```diff
@@ -61,34 +62,29 @@ public ResourcePool getPoolForResource(ResourceType resourceType) {
    */
   @Override
   public ResourcePools validateAndMerge(ResourcePools toBeUpdated) {
+    // Ensure update pool types already exist in existing pools
     if(!getResourceTypeSet().containsAll(toBeUpdated.getResourceTypeSet())) {
       throw new IllegalArgumentException("Pools to be updated cannot contain previously undefined resources pools");
     }
+    // Can not update OFFHEAP
+    // TODO: Move to OFFHEAP ResourcePool
```

Comment:
I think it's better that we file issues for the todos in the non clustered code and remove the todos, this way we have a better chance of tracking and resolving all the todos before unclustered GA.